spec/push/message_apns_spec.rb

```diff
@@ -0,0 +1,135 @@
+require 'spec_helper'
+
+describe Push::MessageApns do
+  let(:notification_options) {
+    {
+      app: 'com.test.app',
+      device: 'ce8be6272e43e85516033e24b4c289220eeda4879c477160b2545e95b68b5969',
+      expiry: 1.day.to_i,
+      badge: 1,
+      attributes_for_device: { test: 1 },
+      sound: 'default',
+      alert: 'We have a new push notification for you',
+      content_available: 1,
+      priority: 10
+    }
+  }
+
+  subject { Push::MessageApns.create(notification_options) }
+
+  describe "#badge" do
+    subject { super().badge }
```

Comment:
Not a fan of super subjects. I would have a let instead of the subject above as `let(:instance) { Push::MessageApns.create(notification_options) }` and refer to `instance` in subjects below